ballerina-tests/tests/test_utils.bal

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/file;
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mime;
+import ballerina/os;
+import ballerina/test;
+
+string sep = file:pathSeparator;
+string currentDir = file:getCurrentDir();
+string rootDir = check file:parentPath(currentDir);
+string ballerinaTestDir = string `${rootDir}${sep}ballerina-tests${sep}tests`;
+string bal = string `${rootDir}${sep}target${sep}ballerina-runtime${sep}bin${sep}bal`;
+
+function runOSCommand(string projName, string projPath, string configFilePath) returns error? {
+    os:Process|os:Error process = os:exec({
```

Comment:
Need to test this on windows as well.
We usually do this kind of (invoking ballerina compiler for test purposes) form gradle script.
Maybe this is also OK, please think about whether this has any drawbacks compared to Gradle approach.

Reply:
Using gradle script for this is not working, since the test services are only running until the test module finish the execution

Reply:
But +1 for the windows, I will separately create a issue to restructure the tests